services/QuillLMS/app/models/unit.rb

```diff
@@ -55,6 +58,12 @@ class Unit < ApplicationRecord
   # on save or touch, and touch explicitly bypasses after_save hooks
   after_commit :touch_all_classrooms_and_classroom_units
 
+  def set_active_to_false
```

Comment:
I think the word `hide` when used in cases not referring to `active` is confusing.  In those cases replace it with `archive`:

```
  def hide_unit_if_archived
    return if visible
  
    self.active = false
  end

  def archive_if_no_visible_unit_activities
    return if unit_activities.exists?(visible: true)

    update(visible: false)
  end

  def archive_classroom_units_and_unit_activities
    return if visible

    unit_activities.where(visible: true).update(visible: false)
    classroom_units.where(visible: true).update(visible: false)
  end

```